internal/fs/fs.go

```diff
@@ -2077,13 +2077,26 @@ func (fs *fileSystem) WriteFile(
 	defer in.Unlock()
 
 	// Serve the request.
-	if err := in.Write(ctx, op.Data, op.Offset); err != nil {
-		return err
+	if fs.shouldTriggerBufferEnabledWrite(in) {
```

Comment:
for local inodes, we are creating temp file in createFile flow: https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/internal/fs/fs.go#L1491C34-L1491C34

This is to cover scenarios where setInodeAttributes is called after createFile and before writeFile. At this point of time, its not clear how will you handle setInodeAttributes, add a TODO atleast saying we need to move to createLocalFile if setInodeattributes doesn't work.

Reply:
As discussed over chat, created an empty buffer object at the time of create file operation and allocated memory to it before write.
Also added a TODO for mtime attribute.